test/Dapr.Client.Test/StateApiTest.cs

```diff
@@ -999,6 +999,49 @@ public async Task DeleteBulkStateAsync_ValidateRequest()
             envelope.States[0].Metadata.Should().ContainKey("partitionKey");
         }
 
+        [Fact]
+        public async Task QueryStateAsync_ValidateResult()
+        {
+            await using var client = TestClient.CreateForDaprClient();
+
+            var queryJson = "{'query':{'filter':{ 'EQ': {'value':'test'}}}}";
+            var stateRequest = new StateQueryRequest()
+            {
+                StoreName= "testStore",
+                Query = queryJson,
+            };
+            var request = await client.CaptureGrpcRequestAsync(async daprClient =>
+            {
+                await daprClient.QueryStateAsync<string>(stateRequest);
+            });
+
+            request.Dismiss();
```

Comment:
See feedback on configuration PR - we can test more of the surface area by including a response.